stdlib/tests/crypto/rpo.rs

```diff
@@ -274,3 +274,30 @@ fn test_squeeze_digest() {
 
     build_test!(even_words, &[]).expect_stack(&even_hash);
 }
+
+#[test]
+fn test_hash_memory() {
+    let compute_inputs_hash = "
+    use.std::crypto::hashes::rpo
+
+    begin
+        push.1.2.3.4.1000 mem_storew dropw
+        push.5.6.7.8.1001 mem_storew dropw
+        push.9.10.11.12.1002 mem_storew dropw
+        push.13.14.15.0.1003 mem_storew dropw
+
+
+        push.15.1000
+
+        #######################################################
+
+        exec.rpo::hash_memory
+    end
+    ";
+
+    #[rustfmt::skip]
+    let expected_hash: Vec<u64> = build_expected_hash(&[
+        1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15
+    ]).into_iter().map(|e| e.as_int()).collect();
+    build_test!(compute_inputs_hash, &[]).expect_stack(&expected_hash);
```

Comment:
Could we also add a couple more test cases:

1. Hashing exactly 8 elements.
2. Hashing fewer than 8 elements (e.g., 6).

Reply:
Ah, my bad, I tested this proc on all variations from 0 to 16 elements, but forget to include it into the final test. I'll fix it.